### PR TITLE
[AMLII-2183] Logs From Integrations Rotation E2E Test and Yaml Template

### PR DESCRIPTION
**Pull request by soberpeach**
### What does this PR do?

Ensures all logs are read for logs from integrations after there are file rotations. By setting the maximum log file size to 1 megabyte and each log to 256 kilobytes, we can ensure that after five log writes there will be at least one rotation. Each log has a counter variable included that is incremented by 1 in each iteration. This count is checked on each iteration to ensure no logs are dropped.

It also adds a method to generate YAML check configs programmatically in the e2e test.

### Motivation

E2E test coverage for logs from integrations.

### Describe how to test/QA your changes

Make sure e2e tests pass.

### Possible Drawbacks / Trade-offs

N/A

### Additional Notes
There will be more e2e tests written for this feature soon, including a test for multiple custom checks, and adding windows tests.

### CODE REVIEW

File: test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go [hunk 1]
```diff
@@ -7,40 +7,145 @@ package integrationslogs
...
+				regex := regexp.MustCompile(`counter: (\d+)`)
+				matches := regex.FindStringSubmatch(log.Message)
+				assert.Greater(v.T(), len(matches), 1, "Did not find count in log")
+
+				number := matches[1]
+				count, err := strconv.Atoi(number)
+				assert.Nil(v.T(), err)
+
+				if prevLogCount != -1 {
+					assert.Equal(v.T(), prevLogCount+1, count)
```
Comment (carlosroman) [inline test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go:118]:
```suggestion
					assert.Equal(c, prevLogCount+1, count)
```

File: test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go [hunk 2]
```diff
@@ -7,40 +7,145 @@ package integrationslogs
...
+
+			if assert.NotEmpty(c, logs) && len(logs) >= i+1 {
+				log := logs[i]
+				regex := regexp.MustCompile(`counter: (\d+)`)
+				matches := regex.FindStringSubmatch(log.Message)
+				assert.Greater(v.T(), len(matches), 1, "Did not find count in log")
+
+				number := matches[1]
+				count, err := strconv.Atoi(number)
+				assert.Nil(v.T(), err)
```
Comment (carlosroman) [inline test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go:115]:
```suggestion
				assert.Nil(c, err)
```

File: test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go [hunk 3]
```diff
@@ -7,40 +7,145 @@ package integrationslogs
...
+	for i := 0; i < 5; i++ {
+		assert.EventuallyWithT(v.T(), func(c *assert.CollectT) {
+			logs, err := utils.FetchAndFilterLogs(v.Env().FakeIntake, "rotation_service", ".*counter: \\d+.*")
+			assert.NoError(c, err)
+
+			if assert.NotEmpty(c, logs) && len(logs) >= i+1 {
+				log := logs[i]
+				regex := regexp.MustCompile(`counter: (\d+)`)
+				matches := regex.FindStringSubmatch(log.Message)
+				assert.Greater(v.T(), len(matches), 1, "Did not find count in log")
```
Comment (carlosroman) [inline test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go:111]:
```suggestion
				assert.Greater(c, len(matches), 1, "Did not find count in log")
```
Or even better use [assert.NotEmpty](https://pkg.go.dev/github.com/stretchr/testify/assert#NotEmpty).

File: test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go [hunk 4]
```diff
@@ -7,40 +7,145 @@ package integrationslogs
...
+	tags := []string{"foo:bar", "env:dev"}
+	yamlData, err := generateYaml("Custom log message", false, 1, 10, tags, "logs_from_integrations_source", "logs_from_integrations_service")
+	assert.NoError(v.T(), err)
+
+	v.UpdateEnv(awshost.Provisioner(awshost.WithAgentOptions(
+		agentparams.WithLogs(),
+		agentparams.WithFile("/etc/datadog-agent/conf.d/writeTenLogs.yaml", string(yamlData), true),
+		agentparams.WithFile("/etc/datadog-agent/checks.d/writeTenLogs.py", customIntegration, true))))
+
+	utils.CheckLogsExpected(v.T(), v.Env().FakeIntake, "logs_from_integrations_service", "", tags)
```
Comment (carlosroman) [inline test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go:77]: No check to see if you got at least ten logs.

**Review by carlosroman — CHANGES_REQUESTED** [review]

File: test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go (hunk 1 above)
Comment (soberpeach) [reply test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go:118]: Changed!

File: test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go (hunk 2 above)
Comment (soberpeach) [reply test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go:115]: changed!

File: test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go (hunk 3 above)
Comment (soberpeach) [reply test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go:111]: changed!

File: test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go (hunk 4 above)
Comment (soberpeach) [reply test/new-e2e/tests/agent-metrics-logs/log-agent/linux-log/integrations/logs_test.go]: Changed!

**Review by carlosroman — APPROVED** [review]

**Pull request merged**